pkg/collector/collector.go

```diff
@@ -308,7 +308,12 @@ func (c *Collector) collect(ch chan<- prometheus.Metric, sCh chan *collectorapi.
 		log.Error(err, "failed to collect container metrics")
 	}
 
-	for _, sc := range c.sideCarConfigs {
+	sccfg, err := c.getSideCarConfigs()
```

Comment:
i think this code would be about 200% clearer if `getSideCarConfigs` was called `getSideCarMetrics` - since there's a field of `c` that has the name `sideCarConfigs`, which isn't involved in the `getSideCarConfigs` method at all.